### DISCUSSION TITLE
[Help]: @container based breakpoints

### DISCUSSION THREAD

**Discussion by mufftee** [davidjerleke/embla-carousel#889] [Help] [answered]
### Summary

Is there a way to use breakpoints based on the container via "(min-width: 320px)"?

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment]
@mufftee I’m not sure I understand what you want to achieve? Do you want to set breakpoints on the container element instead of window?

**Reply by mufftee** [reply]
Exactly.

**Reply by mufftee** [reply]
It's "pretty new" but almost supported by all browsers if i am not wrong

[here is a link to the @container queries](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_containment/Container_queries)

**Reply by davidjerleke** [reply, marked as answer]
@mufftee then it’s probably a no. Not familiar with the @container CSS rule so I might be wrong though. The `breakpoints` option is using the browser native `window.matchMedia()` under the hood so any limitations to that method applies.

However, you can pass a custom callback to the `watchResize` option and check if the element is equal to the container. If yes, you can do stuff based on the container width.

**Reply by mufftee** [reply]
Sorry to be that lazy, do you have an example for that laying around?

**Reply by davidjerleke** [reply]
@mufftee like this:

```tsx
const options = {
  watchResize: (emblaApi, entries) => {
    entries.forEach(entry => {
      if (entry.target === emblaApi.containerNode()) {
        // The container size has changed, add your logic and do something here
      }
    })
  }
} 
```

**Comment by mufftee** [comment]
If somebody wants to take a deep dive into this topic

W3C: The [CSSContainerRule](https://www.w3.org/TR/css-contain-3/#csscontainerrule) interface represents a [@container](https://www.w3.org/TR/css-contain-3/#at-ruledef-container) rule.

[Github Discussion to this topic](https://github.com/w3c/csswg-drafts/issues/6205)
